src/main.cpp

```diff
@@ -905,7 +910,20 @@ static int main0(int argc, char **argv) {
         }
         if (c_source_files.length == 0 && objects.length == 0) {
             // For example `zig cc` and no args should print the "no input files" message.
-            return ZigClang_main(argc, argv);
+            if(zig_only_argc) {
+                // const char **argv_;
+                ZigList<const char *> clang_argv = {0};
+                for (int i = 0; i < argc; i += 1) {
+                    if(strcmp(argv[i], "--cache-dir") == 0) {
+                        i += 1;
+                    } else {
+                        clang_argv.append(argv[i]);
+                    }
+                }
+                return ZigClang_main(clang_argv.length, (char **)clang_argv.items);
```

Comment:
this is the part I don't like. could someone point me to the correct way to do this?